src/dxtbx/nexus/nxmx.py

```diff
@@ -490,10 +496,8 @@ def depends_on(self) -> NXtransformationsAxis | None:
     @cached_property
     def temperature(self) -> pint.Quantity | None:
         """The temperature of the sample."""
-        if "temperature" in self._handle:
-            temperature = self._handle["temperature"]
-            units = h5str(temperature.attrs["units"])
-            return temperature[()] * ureg(units)
+        if temperature := self._handle.get("temperature"):
+            return temperature[()] * units(temperature)
```

Comment:
Does this correctly distinguish between absence (i.e. `self._handle.get("temperature") == None` and non-truthiness? I suspect a h5py dataset containing a zero value _is_ "truthy", but it would be good to confirm this. If so it might be useful to add a test verifying that we handle this correctly.

Reply:
Yes, exactly, an `h5py.Dataset` is truthy, even if its content data is not.

I agree a test would be valuable, I'll add one.

Reply:
Done.